tests/testautotls.nim

```diff
@@ -0,0 +1,45 @@
+{.used.}
+
+# Nim-Libp2p
+# Copyright (c) 2023 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+{.push raises: [].}
+
+# import std/[strformat, net] # uncomment after re-enabling AutoTLSManager
+import chronos
+import chronos/apps/http/httpclient
+import ../libp2p/[stream/connection, upgrademngrs/upgrade, autotls/acme/api, wire]
+
+import ./helpers
+
+suite "AutoTLS":
+  teardown:
+    checkTrackers()
+
+  asyncTest "test ACME":
+    let api = await ACMEApi.new(acmeServerURL = LetsEncryptURLStaging)
+    defer:
+      await api.session.closeWait()
```

Comment:
`await api.session.closeWait()`
could this be 
`await api.close()`

usually code that creates something (in this case ACMEApi instance) should not access individual properties to close, but rather type should expose method to close all resources